tekton/pipeline_run.go

```diff
@@ -101,6 +103,19 @@ func (r *ReleasePipelineRun) WithReleaseLabels(releaseName, releaseNamespace str
 	return r
 }
 
+// WithApplicationSnapshot adds a param containing the ApplicationSnapshot as a json string
+// to the release PipelineRun.
+func (r *ReleasePipelineRun) WithApplicationSnapshot(snapshot *v1alpha1.ApplicationSnapshot) *ReleasePipelineRun {
+	snapshotString, _ := json.Marshal(snapshot.Spec)
+
+	r.WithExtraParam(snapshot.Kind, tektonv1beta1.ArrayOrString{
+		Type:      tektonv1beta1.ParamTypeString,
+		StringVal: string(snapshotString),
```

Comment:
This line is WIP because tekton does crazy things with string variables

Reply:
I converted this PR into a draft as per your comment.